lib/yargs-factory.ts

```diff
@@ -1044,7 +1044,11 @@ export class YargsInstance {
       }
 
       const desc = opt.describe || opt.description || opt.desc;
-      this.describe(key, desc);
+      const existingDesc = this.#usage.getDescriptions()[key];
+      if (!existingDesc || typeof desc === 'string') {
```

Comment:
nit:

We've had bugs pop up in the past because we're checking truthiness of the value rather than if the key is set or not.

Could it be written this way instead?

```js
const descriptions = this.#usage.getDescriptions();
if (
  !Object.prototype.hasOwnProperty.call(descriptions, key) ||
  typeof desc === 'string'
) {
  this.describe(key, desc);
}
```